pkg/server/misc.go

```diff
@@ -0,0 +1,751 @@
+	getListAndEncodeToYAML("cronjobs", c.GetAllCronJobsList, dir, errLog)
+	getListAndEncodeToYAML("nodes", c.GetAllNodesList, dir, errLog)
+	getListAndEncodeToYAML("configmaps", c.GetAllConfigMaps, dir, errLog)
+	getListAndEncodeToYAML("volumeattachments", c.GetAllVolumeAttachments, dir, errLog)
+}
+
+func (s *HarvesterServer) generateSupportBundleYAMLsForHarvester(dir string, errLog io.Writer) {
+	clientset, err := versioned.NewForConfig(s.RESTConfig)
+	if err != nil {
+		logrus.Error(err)
+		return
+	}
+
+	// Harvester
+	for _, ns := range []string{s.namespace, "default"} {
+		hc := NewHarvesterClient(s.Context, ns, clientset)
+		toDir := filepath.Join(dir, "harvester", ns)
+		getListAndEncodeToYAML("keypairs", hc.GetAllKeypairs, toDir, errLog)
+		getListAndEncodeToYAML("preferences", hc.GetAllPreferences, toDir, errLog)
+		getListAndEncodeToYAML("settings", hc.GetAllSettings, toDir, errLog)
+		getListAndEncodeToYAML("upgrades", hc.GetAllUpgrades, toDir, errLog)
+		getListAndEncodeToYAML("users", hc.GetAllUsers, toDir, errLog)
+		getListAndEncodeToYAML("virtualmachinebackups", hc.GetAllVirtualMachineBackups, toDir, errLog)
+		getListAndEncodeToYAML("virtualmachinebackupcontents", hc.GetAllVirtualMachineBackupContents, toDir, errLog)
+		getListAndEncodeToYAML("virtualmachineimages", hc.GetAllVirtualMachineImages, toDir, errLog)
+		getListAndEncodeToYAML("virtualmachinerestores", hc.GetAllVirtualMachineRestores, toDir, errLog)
+		getListAndEncodeToYAML("virtualmachinetemplates", hc.GetAllVirtualMachineTemplates, toDir, errLog)
+		getListAndEncodeToYAML("virtualmachinetemplateversions", hc.GetAllVirtualMachineTemplateVersions, toDir, errLog)
+	}
+
+	// KubeVirt & CDI
+	ns := "default"
+	hc := NewHarvesterClient(s.Context, ns, clientset)
+	toDir := filepath.Join(dir, "kubevirt", ns)
+	getListAndEncodeToYAML("virtualmachines", hc.GetAllVirtualMachines, toDir, errLog)
+	getListAndEncodeToYAML("virtualmachineinstances", hc.GetAllVirtualMachineInstances, toDir, errLog)
+	getListAndEncodeToYAML("virtualmachineinstancemigrations", hc.GetAllVirtualMachineInstanceMigrations, toDir, errLog)
+
+	toDir = filepath.Join(dir, "cdi", ns)
+	getListAndEncodeToYAML("datavolumes", hc.GetAllDataVolumes, toDir, errLog)
+}
+
+func getListAndEncodeToYAML(name string, getListFunc GetRuntimeObjectListFunc, yamlsDir string, errLog io.Writer) {
+	obj, err := getListFunc()
+	if err != nil {
+		fmt.Fprintf(errLog, "Support Bundle: failed to get %v: %v\n", name, err)
+	}
+	encodeToYAMLFile(obj, filepath.Join(yamlsDir, name+".yaml"), errLog)
+}
+
+func encodeToYAMLFile(obj interface{}, path string, errLog io.Writer) {
+	var err error
+	defer func() {
+		if err != nil {
+			fmt.Fprintf(errLog, "Support Bundle: failed to generate %v: %v\n", path, err)
+		}
+	}()
+	err = os.MkdirAll(filepath.Dir(path), os.FileMode(0755))
+	if err != nil {
+		return
+	}
+	f, err := os.Create(path)
+	if err != nil {
+		return
+	}
+	defer f.Close()
+	encoder := yaml.NewEncoder(f)
```

Comment:
I found [an encoding issue](https://github.com/longhorn/longhorn/issues/2490) in Longhorn SB. We might need to use a serializer in Kubernetes library. I haven't validated but you may try:
```
k8s.io/apimachinery/pkg/runtime/serializer/json.NewSerializerWithOptions(
    k8sJson.DefaultMetaFactory, nil, nil,
    k8sJson.SerializerOptions{
        Yaml:   true,
        Pretty: true,
        Strict: true,
    },
)
```

Reply:
[Fixed in the manager code](https://github.com/harvester/support-bundle-utils/pull/1/commits/3b7110886b29319207ef50c74670f6892cdd7492#diff-db8a1eef39641ac9a0a2e67b8cf01716633b428c614186ce6cd280d3a73edaaaR160-R179) and it works, thanks!